pkg/queues/amqp_provider.go

```diff
@@ -0,0 +1,190 @@
+package queues
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"sync/atomic"
+
+	"github.com/flightctl/flightctl/pkg/log"
+	"github.com/flightctl/flightctl/pkg/reqid"
+	"github.com/go-chi/chi/v5/middleware"
+	amqp "github.com/rabbitmq/amqp091-go"
+	"github.com/sirupsen/logrus"
+)
+
+type amqpProvider struct {
+	doneCh  chan struct{}
+	url     string
+	log     logrus.FieldLogger
+	wg      *sync.WaitGroup
+	queues  []*amqpQueue
+	stopped atomic.Bool
+}
+
+func NewAmqpProvider(url string, log logrus.FieldLogger) Provider {
+	var wg sync.WaitGroup
+	wg.Add(1)
```

Comment:
drop

Reply:
I want to be able to wait on provider even if there are no consumer loops yet.  How can I achieve it without it?

Reply:
For example, I would like the following code to run properly:
```
   provider := queues.NewAmqpProvider(url, log)
   ...
   // Wait until provider is stopped
   provider.Wait() 
```

Reply:
So to be clear what you are doing would technically work it is just not proper usage of a wait.Group. Wait group is intended for waiting for goroutines to finish[1]. So if you wanted to just block with no goroutines you have a couple of options. Add back your doneCh or wait on the ctx.

```go
func (a *amqpProvider) Wait() {
        <-a.doneCh
	a.wg.Wait()
}
```
```go
func (a *amqpProvider) Wait(ctx context.Context) {
        <-ctx.Done()
	a.wg.Wait()
}
```

[1]  https://pkg.go.dev/sync#WaitGroup

Reply:
As we discussed offline change here is not necessary.